legal-api/src/legal_api/services/queue.py

```diff
@@ -120,17 +120,25 @@ async def close(self):
     def publish_json(self, payload=None):
         """Publish the json payload to the Queue Service."""
         try:
-            self.loop.run_until_complete(self.async_publish_json(payload))
+            self.loop.run_until_complete(self.async_publish_json(payload, self.subject))
         except Exception as err:
             self.logger.error('Error: %s', err)
             raise err
 
-    async def async_publish_json(self, payload=None):
+    async def publish_json_to_subject(self, payload=None, subject=None):
```

Comment:
Added a new method here because the self.subject is the filer one and it is better not to disturb that. Also the existing publish_json calls run_until_complete which throws an error  in this case as the event loop is already running from the update_legal_filings.py. We just have to use an async/await here

Reply:
Nicely done!